bin/app-icon.js

```diff
@@ -95,38 +89,31 @@ program
   .option('-o, --output <output>', "The output image, .e.g 'icon-out.png'.")
   .option('-t, --top [top]', "The label to put on the top of the image, .e.g 'qa'.")
   .option('-b, --bottom [bottom]', "The label to put on the bottom of the image, .e.g '1.2.5'.")
-  .action((parameters) => {
+  .action(async (parameters) => {
     const {
       input,
       output,
       top,
       bottom,
     } = parameters;
 
-    imagemagickCli.getVersion()
-      .then((version) => {
-        if (!version) {
-          console.error('  Error: ImageMagick must be installed. Try:');
-          console.error('    brew install imagemagick');
-          return process.exit(1);
-        }
-
-        //  Check that we have a input file.
-        return fileExists(input);
-      })
-      .then((exists) => {
-        if (!exists) {
-          console.error(`Input file '${input}' does not exist.`);
-          return process.exit(1);
-        }
-        //  Generate some icons then innit.
-        return labelImage(input, output, top, bottom);
-      })
-      .catch((labelErr) => {
-        console.error('An error occurred labelling the icon...');
-        console.log(labelErr);
-        return process.exit(1);
-      });
+    const version = await imagemagickCli.getVersion();
+
+    if (!version) {
+      console.error('  Error: ImageMagick must be installed. Try:');
+      console.error('    brew install imagemagick');
+      return process.exit(1);
+    }
+
+    await errorIfMissing(input, `Input file '${input}' does not exist.`);
+
+    try {
+      await labelImage(input, output, top, bottom);
+    } catch (err) {
+      console.error('An error occurred labelling the icon...');
```

Comment:
Might be easier to just do:

```js
console.error('An error occurred labelling the icon: ${err}');
```

Another improvement would be to perhaps standardise the error logging a little, to make sure we print in colour, using `chalk`, and perhaps even only show the internal error message if a verbose flag like `-v` is set - but this can happen later.

It also means we could potentially simplify things with something like:

```
function errorAndExit(errorMessage, internalError, verbose, exitCode) {
   console.error(`${chalk.red('error')}: ${errorMessage}`);
   if (verbose) {
     console.error(`${chalk.red('internal error')}: ${(internalError || '').toString()}`);
   }
   process.exit(exitCode);
}
```

Also really I should document in the `README` what the error codes are and actually use separate error codes for separate conditions, but that's an improvement for another time.

Reply:
If you log the error by stringifying it, you will omit the stacktrace. Perhaps that's something you'd like to include?

Here is an example:
```js
try {
  throw Error('example error message')
} catch(err) {
  console.error(`An error occurred labelling the icon: ${err}`);
}
```
![image](https://user-images.githubusercontent.com/13087421/56862801-61bf5d00-69e1-11e9-8456-306cc03d876d.png)

vs.
```js
try {
  throw Error('example error message')
} catch(err) {
  console.error('An error occurred labelling the icon:', err);
}
```
![image](https://user-images.githubusercontent.com/13087421/56862784-418f9e00-69e1-11e9-8a74-46e8111b2f06.png)

or

```js
try {
  throw Error('example error message')
} catch(err) {
  console.error(err, 'An error occurred labelling the icon:');
}
```
![image](https://user-images.githubusercontent.com/13087421/56862762-0db47880-69e1-11e9-98a0-de573ac91a4b.png)

Reply:
Good point, makes me more in favour of a single unified `handleError` function of some kind, which prints the error, includes the stacktrace if verbose mode is specified, and colourises the output a little to make it cleaner for the user...

Reply:
Lets raise a seperate ticket for error handling, as there is quite a lot of refactoring to do.

Would also need to factor in cases when its used as a CLI vs as a Node module where you have 2 seperate use cases for how to throw and display error messaging

Reply:
Raised #103